src/drivers/boards/sitl/board_config.h

```diff
@@ -39,8 +39,8 @@
 
 #pragma once
 
-#define BOARD_OVERRIDE_UUID "SIMULATIONID" // must be of length 12 (PX4_CPU_UUID_BYTE_LENGTH)
-#define BOARD_OVERRIDE_MFGUID BOARD_OVERRIDE_UUID
+#define BOARD_OVERRIDE_UUID "SIMULATIONID0000" // must be of length 16
+#define PX4_SOC_ARCH_ID 0x1003
```

Comment:
I'm wondering if we should create an enum at a common place to list all arch ID's, so that we never use the same ID on 2 different archs?

Reply:
@bkueng - Yes we should (most likely in px4_microhal.h). But not being familiar with the non nuttx board I am at a deficit here.  The original macros were just placeholders. Now the API is in place but adds no value. It really needs to be located per board or arch as a module.  I think that there needs to be real implementations for the Board Identity API in all the other boards/arch. When that is done we can remove all the hacks in the non nuttx src/drivers/boards/*/board_config.h and system_config.h 
 
Who best knows the other boards and can implement the Board Identity API i for them?

Reply:
I agree. We can do that in steps - but I think it comes down to either you or me doing it :)
To get an ID on RPi I found this: https://www.raspberrypi-spy.co.uk/2012/09/getting-your-raspberry-pi-serial-number-using-python/. And on other boards we could use the MAC if they don't have a CPU ID and provide an internal network interface.

Reply:
@mcharleb - how would we get the UUID on snappy?